paparazzi-gradle-plugin/src/main/java/app/cash/paparazzi/gradle/PrepareResourcesTask.kt

```diff
@@ -65,6 +65,8 @@ open class PrepareResourcesTask : DefaultTask() {
           it.newLine()
           it.write(platformDataRoot.get().asFile.path)
           it.newLine()
+          it.write("build/intermediates/compile_and_runtime_not_namespaced_r_class_jar/debugUnitTest")
```

Comment:
I searched for a property that would give me this but didn't find anything. Hopefully there's a better way.

Reply:
I'm not entirely sure how this task works, but hard-coding a path like this is, generally, not safe. For one thing, note the terminal element, `debugUnitTest`. Does that mean this task only works on debug build? What about flavors? Other build types?

Reply:
I have some Gradle snippet code that i typically use to analyze/print task inputs and outputs over which i would then grep, to find the appropriate task output dir to which I'd then wire the plugin task.

Afk at the moment, but can paste what I mean later!

Reply:
@autonomousapps Agreed. Don't want to merge with this just trying to get the conversation started because I wasn't able to find an API that gave me what we want here. Will keep digging while I wait on @jrodbx's snippet.

Reply:
I managed to piece together the path using an object that I found looking through AGP's source code and the unit test variant name. The only thing that I'm not excited about is how the object that I'm using is in an internal package. 
Probably not a great idea to depend on it. 🙁

Reply:
I think the new snippet is closer, but what you really want is to depend on that task and feed its outputs into this task's inputs. In fact, you kind of have to, because I think this would fail in Gradle 8 -- they're getting more strict. On the plus side, if I'm right, if you run this code with Gradle 7, then it will tell you which task you're stepping on and that would be a big hint what to use.

Reply:
Ok, in a past talk, I offered some hacks/debugging tips to introspect what tasks get added by AGP (or any plugin, really):
https://speakerdeck.com/jrodbx/deep-er-dive-into-the-android-gradle-plugin-droidcon-sf-2018?slide=74
https://speakerdeck.com/jrodbx/deep-er-dive-into-the-android-gradle-plugin-droidcon-sf-2018?slide=76

By adding this to `paparazzi/build.gradle`:
```
gradle.taskGraph.whenReady {
  gradle.taskGraph.allTasks.each { task ->
    println "task: " + task.name + ", class: " + task.class.name
  }
}
gradle.taskGraph.afterTask { task ->
  println "  inputs: "
  task.inputs.files.each { println "    $it" }
  println "  outputs: "
  task.outputs.files.each { println "    $it" }
}
```
and running `./gradlew sample:test --console=plain`, 
then searching for `compile_and_runtime_not_namespaced_r_class_jar` (given how unique that segment is) and finding the task that outputs something at that path:

```
> Task :sample:generateDebugUnitTestStubRFile
  ...
  outputs:
    /Users/jrod/Development/paparazzi/sample/build/intermediates/compile_and_runtime_not_namespaced_r_class_jar/debugUnitTest/R.jar
```

then searching for `generateDebugUnitTestStubRFile`, you'll find the task class itself (ignoring the _Decorated part, that's a Gradle thing)!:
```
task: generateDebugUnitTestStubRFile, class: com.android.build.gradle.internal.res.GenerateLibraryRFileTask_Decorated
```

Alternatively, if you're familiar with AGP internals, the variant is "DebugUnitTest" so the task name is usually a concatenation of the variant with a prefix and/or suffix, i.e., "generate${VARIANT}StubRFile".

So either: 
https://github.com/jrodbx/agp-sources/search?q=StubRFile
or
https://github.com/jrodbx/agp-sources/search?q=GenerateLibraryRFileTask

will get you to some version of:
https://github.com/jrodbx/agp-sources/blob/a9517caa493219cfea2360f6ebb84daf5a5505f3/7.0.0-alpha07/com.android.tools.build/gradle/com/android/build/gradle/internal/res/GenerateLibraryRFileTask.kt

Looking at that file, you'll see things like:
```
    @get:OutputFile
    abstract val rClassOutputJar: RegularFileProperty
```
and
```
          creationConfig.artifacts.setInitialProvider(
                taskProvider,
                GenerateLibraryRFileTask::rClassOutputJar
            ).withName("R.jar").on(InternalArtifactType.COMPILE_AND_RUNTIME_NOT_NAMESPACED_R_CLASS_JAR)
```

Now, in the sample on master, there's only 1 R.jar.  But in the change proposed by this PR, i'm guessing there will be multiple GenerateLibraryRFileTask instances, each of which will produce an R.jar.

Looking at the current plugin code, there are some exposed TaskProviders on the variant type (i.e., BaseVariant); however, none seem to match the type we're looking for.  However, BaseVariant does expose `DomainObjectCollection<BaseVariantOutput> getOutputs()` and `BaseVariantOutput` exposes:

```
    /**
     * Returns the {@link TaskProvider} for the Android Resources processing task.
     *
     * <p>Prefer this to {@link #getProcessResources()} as it triggers eager configuration of the
     * task.
     */
    @NonNull
    TaskProvider<ProcessAndroidResources> getProcessResourcesProvider();
```

so maybe in the variant block, something like:
```
      val libraryRFileTaskProviders = variant.outputs.map { it.processResourcesProvider as TaskProvider<GenerateLibraryRFileTask> }
```
will finallly give you access to:

```
    @get:OutputFile
    abstract val rClassOutputJar: RegularFileProperty
```
or maybe?
```
    @get:Input
    abstract val packageForR: Property<String>
```

Reply:
Honestly I love that. Very cool process. Your point about artifacts is a good reminder that recent AGPs are starting to expose their artifacts directly, via either `android` or `androidComponents`. Might be worth checking what's available there.